test/GhSyncTests/MockStartup.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace gh_sync.Tests;
+
+public class MockStartup
+{
+    private readonly Lazy<IServiceProvider> services;
+    public IServiceProvider Services => services.Value;
+
+    public MockStartup()
+    {
+        this.services = new Lazy<IServiceProvider>(() =>
+        {
+            var serviceCollection = new ServiceCollection();
+            this.ConfigureServices(serviceCollection);
+            return serviceCollection.BuildServiceProvider();
+        });
+    }
+
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.AddSingleton<IAdo>(
```

Comment:
Especially for testing how synchronization works, this will need fleshed out to specify what different methods should return when called on the mock object.